test/libsolidity/syntaxTests/freeFunctions/free_identical.sol

```diff
@@ -0,0 +1,2 @@
+function f() pure returns (uint) { return 1337; }
+function f() pure returns (uint) { return 42; }
```

Comment:
This should definitely not be allowed - maybe overloading, but definitely not a plain redefinition...
I'm rather surprised that this didn't have tests...